pyanaconda/modules/services/services.py

```diff
@@ -165,13 +172,23 @@ def set_disabled_services(self, services):
     @property
     def default_target(self):
         """Default target of the installed system."""
-        return self._default_target
+        # if no target has been set, default to the text only target
+        if not self._default_target:
+            return TEXT_ONLY_TARGET
+        else:
+            return self._default_target
 
     def set_default_target(self, target):
         """Set the default target of the installed system.
 
         :param target: a string with the target
         """
+        if target not in self.SUPPORTED_SYSTEMD_TARGETS:
+            msg = "Unsupported systemd default target: {} Specify one of: {}".format(
+                target, self.SUPPORTED_SYSTEMD_TARGETS
+            )
+            raise SystemdDefaultTargetConfigurationError(msg)
```

Comment:
Please, raise `ValueError` here. There no need for a new exception.

Reply:
Good point, indeed no longer needed if this check is not in a DBUS Task subclass.